include/jemalloc/internal/arena_externs.h

```diff
@@ -47,9 +47,9 @@ edata_t *arena_extent_alloc_large(tsdn_t *tsdn, arena_t *arena,
 void arena_extent_dalloc_large_prep(tsdn_t *tsdn, arena_t *arena,
     edata_t *edata);
 void arena_extent_ralloc_large_shrink(tsdn_t *tsdn, arena_t *arena,
-    edata_t *edata, size_t oldsize);
+    edata_t *edata, size_t oldusize);
```

Comment:
While following this path, I realized that this probably needs to be updated following #2459
https://github.com/jemalloc/jemalloc/blob/dev/src/arena.c#L300
https://github.com/jemalloc/jemalloc/blob/dev/src/arena.c#L316

This shows in a different way how helpful your static-analysis fixes are :)

Reply:
It might not be as straightforward though, since for large stats, i.e. `arena->stats.lstats`, the index starts from SC_LARGE_MINCLASS, therefore the `hindex`. We technically should update arena->bin.stats like here:
https://github.com/jemalloc/jemalloc/blob/dev/src/arena.c#L1175

but we neither hold that mutex at that moment, nor it was really allocated under that mutex. What's your opinion there?

Reply:
I'm not sure if there's any great answer here. If we include it in `bin.stats` that's going to potentially mislead anyone using those stats for memory accounting by under-reporting. If we do what we currently do, that's also wrong, but by over-reporting memory usage. I suppose the "proper" thing to do would be to add a new field:
```c
typedef arena_stats_large_t arena_stats_small_sampled_t;

struct arena_stats_s {
    ...
    arena_stats_small_sampled_t sampled_stats[4];
};
```
But then we have to go through all of the trouble of plumbing that through everywhere.

Reply:
I agree that the current way is over-reporting. For the under-reporting part, not sure if I follow. I'm thinking to include it in the nmalloc of the corresponding bin, e.g. the one for 4K, 8K and 12K which is the actual size used for the sample. However it does feel wrong, because it's not really allocated through the bin mechanism.

Reply:
Oh, okay I see. When I said "under-reporting" I thought we would still be attributing the allocation to the bin for the non-promoted `usize` of the allocation, so that we don't artificially make certain size-classes look hotter than they actually are (i.e. imagine you have _tons_ of 8-byte allocations, enough that if you record the sampled ones as 4 K allocations it makes the 4 K bin look hot, misleading your optimization efforts).

Reply:
Agree that there are two type of stats we should report:
1) what actually happens at the allocator level (this is the case before the prof overhead PR)
2) what the application actually requested (this isn't reported, before or after the prof overhead PR)

After the prof overhead PR, the stats is a bit off because right now it's neither. And if we want to still do (1), there isn't a place to report "small & non-slab" request.

Does the below sound reasonable?
1) pile up the `nmalloc` and `nrequests` into the bin stats of the corresponding page size. But probably not `curregs` because it should match the slab stats. 
2) adding a new counter for prof sampled small requests, either a total or per bin (which will only have non-zeros for page sizes)

Either way that certainly should happen separately. I'll merge this PR first.

Reply:
Addressed by #2486.